lib/src/blockdev.rs

```diff
@@ -80,9 +85,17 @@ pub(crate) struct LoopbackDevice {
     pub(crate) dev: Option<Utf8PathBuf>,
 }
 
+// Ensure that `/dev` directory exists and is mounted.
+fn ensure_dev_mounted() -> Result<()> {
+    crate::mount::ensure_mount(DEVTMPFS, DEV_MOUNT_POINT, DEVTMPFS)?;
+    Ok(())
+}
+
 impl LoopbackDevice {
     // Create a new loopback block device targeting the provided file path.
     pub(crate) fn new(path: &Path) -> Result<Self> {
+        // Ensure /dev exists and is mounted
+        ensure_dev_mounted()?;
```

Comment:
Optional: I'm OK with this, although there's some action-at-a-distance here and arguably we should be doing this very early on in the callers instead (and we could add an assertion here that it's been done).